homeassistant/components/oralb/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from oralb import OralB
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import bluetooth
+from homeassistant.components.sensor import (
+    PLATFORM_SCHEMA,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.const import CONF_MAC, CONF_NAME, PERCENTAGE, TIME_SECONDS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=3)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Required(CONF_MAC): cv.string,
+    }
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    ble_device = bluetooth.async_ble_device_from_address(
+        hass, entry.data[CONF_MAC], connectable=True
+    )
+    _LOGGER.info("OralB setup")
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find OralB device with address {entry.data[CONF_MAC]}"
+        )
+    if ble_device:
+        _LOGGER.info("Found OralB toothbrush")
+        orlb = OralB(ble_device)
+
+    async_add_entities(
+        [
+            Battery(orlb),
+            Status(orlb),
+            BrushTime(orlb),
+            Mode(orlb),
+        ]
+    )
+
+
+class OralBSensor(SensorEntity):
+    """Master class for inheriting on sensors."""
+
+    def __init__(self, orlb):
+        """Initialize a sensor."""
+        self.orlb = orlb
+        self._attr_unique_id = self.orlb.ble_device.address + self._attr_name
+
+    async def async_update(self):
+        """Read data from OralB."""
+        await self.orlb.gatherdata()
```

Comment:
This needs to be done inside an update coordinator or else each entity will make a connection and trigger an update. See this example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/co2signal/__init__.py#L60-L86

Reply:
Once you have an update coordinator, let OralBSensor inherit from both the UpdateCoordinatorEntity and SensorEntity like here: https://github.com/home-assistant/core/blob/dev/homeassistant/components/co2signal/sensor.py#L67-L72

Then make a list of entity descriptions like this: https://github.com/home-assistant/core/blob/dev/homeassistant/components/co2signal/sensor.py#L35

And finally, the native value property can live on `OralBSensor` and be:
```python
@property
def native_value(self):
    value = self.coordinator.data[self.entity_description.key]
    if value == "UNKNOWN":
        value = None
    return value
```